[AArch64]Enable aggressive interleaving for A320

nasherm opened the pull request:
This patch makes use of aggressive interleaving options for the A320 subtarget. This is done by adding a new local parameter to the AArch64Subtarget class. With this enabled we see an aggregate uplift of 0.7% on internal benchmark suites with up to 51% uplift on individual benchmark workloads.

nasherm committed:
[AArch64]Enabled aggressive interleaving for A320

This patch makes use of aggressive interleaving options
for the A320 subtarget. This is done by adding a new local
parameter to the AArch64Subtarget class. With this flag
enabled we see an aggregate uplift of 0.7% on internal benchmark suites
with up to 51% uplift on individual benchmark workloads.

Change-Id: I53367df0aef299e0d02ce9150a105f12fa575b9d

nasherm committed:
Code formatting

Change-Id: I6de3eec3da77cf211b060ed9541f1caee7fe0606

davemgreen reviewed (commented):
Can you make this a subtarget feature and add a test.

I'm a little surprised this is profitable (or worth the codesize gain), but these cores do like extra unrolling.

nasherm committed:
Review comments

- added test for aggresive interleaving codegen on A320
- added SubtargetFeature for AggressiveInterleaving

Change-Id: Idd0c33346223dbe53d3f540d2f38a4028d6eb2e8

nasherm committed:
Update test

Change-Id: Icad4c882f22cd1bbd787c5eedcee27ef74f31b7c

nasherm committed:
Test failures

Change-Id: Id7010a4705058fb89bdb75ee1eb2335453d19447

nasherm commented on llvm/lib/Target/AArch64/AArch64Subtarget.cpp:
```diff
@@ -175,6 +175,8 @@ void AArch64Subtarget::initializeProperties(bool HasMinSize) {
     break;
   case CortexA320:
+    AggressiveInterleaving = true;
+    [[fallthrough]];
```
Done

davemgreen reviewed (approved):
Thanks - LGTM
